rest_framework/serializers.py

```diff
@@ -947,6 +947,11 @@ def get_fields(self):
         # Determine any extra field arguments and hidden fields that
         # should be included
         extra_kwargs = self.get_extra_kwargs()
+        for field in extra_kwargs:
+            assert field not in declared_fields, (
+                "Field {} is declared on the class and also present in "
+                "extra_kwargs or read_only_fields".format(field)
+            )
```

Comment:
On thinking about this: we *might* need to be a bit clever about this - some folks may want to use subclassing to override a declared field. Unclear.